pkg/api/mock.go

```diff
@@ -85,3 +85,19 @@ func NewDebugMock(o io.Writer, res ...mock.Response) *API {
 
 	return api
 }
+
+// NewMockMatchingByEndpoint creates a new api.API from a list of Responses, matching by endpoint.
+// Defaults to a dummy APIKey for authentication, which is not checked
+func NewMockMatchingByEndpoint(res map[string][]mock.Response) *API {
```

Comment:
is it by design that this allows multiple responses to match a single endpoint?

Reply:
Yeah, the idea is that you can mock multiple responses for the same endpoint (or the same set of endpoints that match the key regex) 🙂